### PR TITLE
[Submission] Dispenser Fireworks

### PR DESCRIPTION
**Pull request by AndanteDevs**
Makes dispensers facing down or in a horizontal direction actually shoot the firework that way

### CODE REVIEW

File: gm4_dispenser_fireworks/data/dispenser_fireworks/functions/direction/global.mcfunction [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#called by dispenser_fireworks:direction/<direction>
+
+#set angle tag
+data merge entity @s {ShotAtAngle:1b}
+
+#add to score
+scoreboard players add @s gm4_df_mod_count 1
+
+#tag entity
+tag @s add gm4_df_ignore
```
Comment (misode) [inline gm4_dispenser_fireworks/data/dispenser_fireworks/functions/direction/global.mcfunction]: This tag is already being added in `check_block`, making this command unnecessary

File: gm4_dispenser_fireworks/data/dispenser_fireworks/functions/direction/global.mcfunction [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#global commands for all directions
+#@s - @e[type=minecraft:firework_rocket,tag=!gm4_df_modified] with dispenser checks succeeded
+#called by dispenser_fireworks:direction/<direction>
+
+#set angle tag
+data merge entity @s {ShotAtAngle:1b}
+
+#add to score
+scoreboard players add @s gm4_df_mod_count 1
```
Comment (misode) [inline gm4_dispenser_fireworks/data/dispenser_fireworks/functions/direction/global.mcfunction]: Remove the `global` function and put the nbt in the same merge that applies motion. Copy the scoreboard command in each direction function.

Comment (AndanteDevs) [reply gm4_dispenser_fireworks/data/dispenser_fireworks/functions/direction/global.mcfunction]: I made a `global` function for if something needs to be changed for all directions. It just makes it easier to modify.

Comment (misode) [reply gm4_dispenser_fireworks/data/dispenser_fireworks/functions/direction/global.mcfunction]: I get that it makes it easier, but it's a useless function call. If you put the `{ShotAtAngle:1b}` merge inside each direction and you replace the `function dispenser_fireworks:direction/global` function call with `scoreboard players add @s gm4_df_mod_count 1` it functions exactly the same.

Comment (AndanteDevs) [reply gm4_dispenser_fireworks/data/dispenser_fireworks/functions/direction/global.mcfunction]: Argh, alright then

**Comment by sparks0** [comment]
I'm sort of in favour of merging this with the other firework stuff you have in the works to be honest :)

**Comment by sparks0** [comment]
Decided the functions of this module should be included with the firework assembly module. Closing this PR. Please merge these files with the other module next time you update that PR.

**Pull request closed without merging**